plugins/automation/spec/services/discourse_automation/create_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+RSpec.describe DiscourseAutomation::Create do
+  describe described_class::Contract, type: :model do
+    it { is_expected.to validate_presence_of :script }
+  end
+
+  describe ".call" do
+    subject(:result) { described_class.call(params:, **dependencies) }
+
+    fab!(:admin)
+
+    let(:guardian) { admin.guardian }
+    let(:params) { { script: "post", trigger: "topic" } }
+    let(:dependencies) { { guardian: } }
+
+    context "when user can't create an automation" do
+      fab!(:user)
+      let(:guardian) { user.guardian }
+
+      it { is_expected.to fail_a_policy(:can_create_automation) }
+    end
+
+    context "when data is invalid" do
+      before { params[:script] = nil }
+
+      it { is_expected.to fail_a_contract }
+    end
+
+    context "when everything's ok" do
```

Comment:
Usually we put `run_successfully` as the first example. If anything fails, it will output the state of the steps using the steps inspector.
```suggestion
    context "when everything's ok" do
      it { is_expected.to run_successfully }

```